modules/tensor_mechanics/src/materials/crystal_plasticity/ComputeCrystalPlasticityThermalEigenstrain.C

```diff
@@ -0,0 +1,61 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComputeCrystalPlasticityThermalEigenstrain.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeCrystalPlasticityThermalEigenstrain);
+
+InputParameters
+ComputeCrystalPlasticityThermalEigenstrain::validParams()
+{
+  InputParameters params = ComputeCrystalPlasticityEigenstrainBase::validParams();
+
+  params.addCoupledVar("temperature", "Coupled temperature variable");
+  params.addRequiredParam<std::vector<Real>>(
+      "thermal_expansion_coefficients",
+      "Vector of values defining the constant second order thermal expansion coefficients, "
+      "depending on the degree of anisotropy, this should be of size 1, 3, 6 or 9");
+
+  return params;
+}
+
+ComputeCrystalPlasticityThermalEigenstrain::ComputeCrystalPlasticityThermalEigenstrain(
+    const InputParameters & parameters)
+  : DerivativeMaterialInterface<ComputeCrystalPlasticityEigenstrainBase>(parameters),
+    _temperature(coupledValue("temperature")),
+    _temperature_old(coupledValueOld("temperature")),
+    _ddeformation_gradient_dT(declarePropertyDerivative<RankTwoTensor>(
+        _deformation_gradient_name, getVar("temperature", 0)->name()))
+{
+  try
+  {
+    _thermal_expansion_coefficients.fillFromInputVector(
+        getParam<std::vector<Real>>("thermal_expansion_coefficients"));
+  }
+  catch (const MooseException & e)
+  {
+    paramError(
+        "thermal_expansion_coefficients",
+        "input vector size should be 1, 3, 6 or 9 to fill the thermal expansion coefficients");
+  }
+}
```

Comment:
I know you spent some time on figuring out how to catch 'MooseException`, but you probably do not need this. When a wrong vector is passed into `fillFromInputVector`, a mooseError will be thrown out to tell users the vector size must be 1, 3, 6 or 9. 

Is my understanding correct? :-)

Reply:
The mooseError from `fillFromInputVector ` does not tell the user directly which line of the input is causing the error. If the user has multiple coefficients like this, it would be hard to debug

Therefore, I suppose catching this exception and throw a parameter error relating to the specific material would be better. 

What do you think?

Reply:
I see your point.  Good!